### PR TITLE
feat(ci): manage external contributions using env

### PR DESCRIPTION
**Pull request by sc979**
## Description

Manage external contributions using env

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [ ] 24.04.x
- [x] master

### CODE REVIEW

File: .github/workflows/web.yml [hunk 1]
```diff
@@ -51,6 +51,16 @@ env:
   widgets_directory: centreon/www/widgets
 
 jobs:
+  set-pr-environment:
+    uses: ./.github/workflows/set-pr-environment.yml
+
+  get-version:
```
Comment (sc979) [inline .github/workflows/web.yml:57]: @tuntoja & @kduret , do you see any issue in moving get-version before the over actions ?

File: .github/workflows/get-version.yml [hunk 2]
```diff
@@ -178,19 +189,33 @@ jobs:
...
-              # Check if pull request branch targeting develop
-              if [[ "$GITHUB_BRANCH_BASE_REF_NAME" == "develop" ]] && [[ "$GITHUB_BRANCH_PR_STATE" == "OPEN" ]]; then
+              echo "Github repository context var: $GITHUB_REPOSITORY"
+              if [[ "${{ inputs.pull_request_environment }}" == "internal_contributor" ]]; then
+                echo "Internal contribution."
                 echo "release_cloud=1" >> $GITHUB_OUTPUT
                 echo "release_type=$GITHUB_RELEASE_TYPE" >> $GITHUB_OUTPUT
-              else
-                echo "release_cloud=$GITHUB_RELEASE_CLOUD" >> $GITHUB_OUTPUT
+                # GITHUB_BRANCH_BASE_REF_NAME="$(gh pr view $BRANCHNAME -q .baseRefName --json headRefName,baseRefName,state)"
```
Comment (sc979) [inline .github/workflows/get-version.yml:197]: @tuntoja , what should we do with these comment ?

File: .github/workflows/get-version.yml [hunk 3]
```diff
@@ -178,19 +189,33 @@ jobs:
...
+                # GITHUB_BRANCH_PR_STATE="$(gh pr view $BRANCHNAME -q .state --json headRefName,baseRefName,state)"
+              elif [[ "${{ inputs.pull_request_environment }}" == "external_contributor" ]]; then
```
Comment (sc979) [inline .github/workflows/get-version.yml:199]: @tuntoja I replaced the IS_INTERNAL variable with the env

File: .github/workflows/get-version.yml (hunk 2 above)
Comment (tuntoja) [reply .github/workflows/get-version.yml:191]: they could be removed,, case is already handled in the develop branches (or should be)

File: .github/workflows/docker-stable.yml [hunk 4]
```diff
@@ -15,15 +15,20 @@ on:
...
     needs: [get-version]
     runs-on: ubuntu-22.04
-    if: ${{ !cancelled() && !contains(needs.*.result, 'failure') && !contains(needs.*.result, 'cancelled') && needs.get-version.outputs.stability == 'stable' && github.ref_name == 'master' }}
-
+    if: |
+      !cancelled() &&
+      !contains(needs.*.result, 'failure') &&
+      !contains(needs.*.result, 'cancelled') &&
+      needs.get-version.outputs.stability == 'stable' &&
+      github.ref_name == 'master'
```
Comment (tuntoja) [inline .github/workflows/docker-stable.yml:31]:
```suggestion
      github.ref_name == 'master' &&
      needs.get-version.outputs.pull_request_environment == 'internal_contributor'
```

Comment (sc979) [reply .github/workflows/docker-stable.yml:31]: get-environment

**Comment by sc979** [comment]
replaced with https://github.com/centreon/centreon/pull/5740

**Pull request closed without merging**